04.ls/wc.rb

```diff
@@ -0,0 +1,77 @@
+# frozen_string_literal: true
+
+require 'optparse'
+
+def main
+  options = parse_options
+  input_files =
+    if ARGV.any?
+      ARGV.map do |file|
+        { name: file, text: File.read(file) }
+      end
+    else
+      [{ name: '', text: $stdin.read }]
+    end
+  process_files(options, input_files)
+end
+
+def parse_options
+  opt = OptionParser.new
+  params = { lines: false, words: false, bytes: false }
+  opt.on('-l') { params[:lines] = true }
+  opt.on('-w') { params[:words] = true }
+  opt.on('-c') { params[:bytes] = true }
+
+  opt.parse!(ARGV)
+  params.values.any? ? params : { lines: true, words: true, bytes: true }
+end
+
+def process_files(options, input_files)
```

Comment:
```suggestion
def process_files(input_files, options)
```

オプションは文字通りオプションなので最後の方に持ってくることが多いですね